src/Cardano/Wallet/CoinSelection/Fee.hs

```diff
@@ -341,3 +352,128 @@ computeFee (CoinSelection inps outs chgs) =
+        --  |   | bytes      -- 2 + 32
+        --  |   | attributes -- 1-8
+        --  |   | word8      -- 1
+        --  | word32                      -- 1-4
+
+        -- For random scheme
+        -- output    ---------------------------------- 69-77 (mainnet) & 77-85 (testnet)
+        --  | list len 2                  -- 1
+        --  | sizeOf(address)             -- 67-75
+        --  | word64                      -- 1-8
+
+        -- address   ---------------------------------- 67 (mainnet) & 75 (testnet)
+        --  | list len 2                  -- 1
+        --  | tag 24                      -- 2
+        --  | bytes          --------------- 2 + 48-56
+        --  |   | list len 3 -- 1
+        --  |   | bytes      -- 2 + 43
+        --  |   | attributes -- 1-8
+        --  |   | word8      -- 1
+        --  | word32                      -- 1-4
+        sizeOfTxOut :: Word64 -> Int
+        sizeOfTxOut =
+            let toAdd = case (scheme, net) of
+                    (Sequential, Mainnet) -> 48
+                    (Sequential, Testnet) -> 56
+                    (Random, Mainnet) -> 69
+                    (Random, Testnet) -> 77
+            in (+ toAdd) . fromIntegral . BL.length . toLazyByteString . encodeWord64
+
+        -- tx    ---------------------------------- 6 + (n * 42) + Σ sizeOf(output)
+        --  | list len 3                  -- 1
+        --  | begin                       -- 1
+        --  | n * sizeOf(input)           -- n * 42
+        --  | break                       -- 1
+        --  | begin                       -- 1
+        --  | Σ sizeOf(output)            --  Σ sizeOf(output)
+        --  | break                       -- 1
+        --  | empty attributes            -- 1
+        sizeOfTx :: Int
+        sizeOfTx =
+            let n = length inps
+                coinOuts = map (getCoin . coin) outs
+                coinChns = map getCoin chngs
+                coins = coinOuts ++ coinChns
+                cborOverhead = 6
+            in cborOverhead + n*sizeOfTxIn + sum (map sizeOfTxOut coins)
+
+        -- witness   ---------------------------------- 139
+        --  | list len 2                  -- 1
+        --  | word8                       -- 1
+        --  | tag 24                      -- 2
+        --  | bytes          --------------- 2 + 133
+        --  |   | list len 2 -- 1
+        --  |   | bytes      -- 2+64
+        --  |   | bytes      -- 2+64
+        sizeOfTxWitness :: Int
+        sizeOfTxWitness = 139
+
+
+        -- With `n` the number of inputs
+        -- signed tx ----------------------------------- 9 + n * 42 + n * 139 + Σ sizeOf(output)
+        --  | list len 2              -- 1
+        --  | sizeOf(tx)              -- 6 + (n * 42) + sizeOf(output)
+        --  | list len n              -- 1-2
+        --  | n * sizeOf(witness)     -- n * 139
+        totalPayload :: Double
+        totalPayload =
+            let n = length inps
+                cborOverhead = 2
```

Comment:
The overhead actually depends on the number of inputs. When there are more than 23 inputs, the length is encoded on 2 bytes (and we can realistically assume that there's no tx with inputs bigger than 255); I'd say that sticking to the upper bound is more conservative here. 

I believe you chose `2` in order to have the property in the unit test to pass. So maybe we could review the property condition to expect that: real fee are lower than or equal to estimated fee, with perhaps also, a predicate on the "error margin" (like, less than 5 Lovelace or so..)

Reply:
leaned towards being precise here and added two test showing the case. hope you dont mind